placement/algo/sharded.go

```diff
@@ -58,25 +58,20 @@ func (a rackAwarePlacementAlgorithm) InitialPlacement(
 		return nil, err
 	}
 
-	p := ph.generatePlacement()
+	var (
+		p   = ph.generatePlacement()
+		err error
+	)
 	for i := 1; i < rf; i++ {
-		ph := newAddReplicaHelper(p, a.opts)
-		if err := ph.placeShards(newShards(p.Shards()), nil, ph.Instances()); err != nil {
-			return nil, err
-		}
-		if err := ph.optimize(safe); err != nil {
+		p, err = a.AddReplica(p)
```

Comment:
Seems we used to only place shards to non leaving instances, I guess that doesn't matter for initial placement?